### PR TITLE
SUBMARINE-60. Remove stubServiceClient from YarnServiceUtils

### PR DESCRIPTION
**Pull request by adamantal**
### What is this PR for?
Remove stubServiceClient from YarnServiceUtils and introduce a Factory based injecting solution. (See jira for further details)

### What type of PR is it?
Refactoring

### Todos
* [ ] - Check whether the refactored test passes.

### What is the Jira issue?
* https://issues.apache.org/jira/browse/SUBMARINE-60

### How should this be tested?
* No need for integration testing, just the UTs.

### Screenshots (if appropriate)
-

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by adamantal** [comment]
@szilard-nemeth Could you please review/submit this PR when you get time?

File: submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/YarnServiceUtils.java [hunk 1]
```diff
@@ -30,35 +30,31 @@
...
-    return componentInstanceName + getDNSNameCommonSuffix(serviceName, userName,
-        domain, port);
+  static AppAdminClient createServiceClient(
+          Configuration yarnConfiguration) {
+    return appAdminClientFactory.createDefault(yarnConfiguration);
   }
 
   public static String getDNSNameCommonSuffix(String serviceName,
-      String userName, String domain, int port) {
+                                              String userName, String domain, int port) {
```
Comment (pingsutw) [inline submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/YarnServiceUtils.java]: should remove extra whitespace

Comment (adamantal) [reply submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/YarnServiceUtils.java]: Done (and also removed extra whitespaces from the whole class).

**Comment by szilard-nemeth** [comment]
Hi @jiwq !
I am a Submarine committer myself, but have never created any commit to this project.
I'd happily commit this. Could you please give me some steps about how to commit to the project?
Read through this guide but I don't see intstructions on how to commit: https://github.com/apache/submarine/blob/master/docs/community/contributing.md
Thanks!

**Comment by jiwq** [comment]
Hi @szilard-nemeth , we use dev-support/cicd/merge_submarine_pr.py to merge PR. Detailed info refer to https://github.com/apache/submarine/tree/master/dev-support/cicd

**Comment by liuxunorg** [comment]
> Hi @jiwq !
> I am a Submarine committer myself, but have never created any commit to this project.
> I'd happily commit this. Could you please give me some steps about how to commit to the project?
> Read through this guide but I don't see intstructions on how to commit: https://github.com/apache/submarine/blob/master/docs/community/contributing.md
> Thanks!

hi @szilard-nemeth , submarine use cicd docker container commit PR, 
you can view commit guide : https://github.com/apache/submarine/tree/master/dev-support/cicd

I am willing to answer any questions. :-)

**Comment by szilard-nemeth** [comment]
Hi @jiwq , @liuxunorg !
Thanks for the steps, followed the guide and merged the PR!
Thanks @adamantal for your contribution!

**Pull request closed without merging**